Assets/MixedRealityToolkit.SDK/Features/Utilities/StabilizationPlaneModifier.cs

```diff
@@ -0,0 +1,390 @@
+
+        /// <summary>
+        /// The active instance of the input system.
+        /// </summary>
+        private IMixedRealityInputSystem InputSystem
+        {
+            get
+            {
+                if (inputSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>(out inputSystem);
+                }
+                return inputSystem;
+            }
+        }
+
+        /// <summary>
+        /// Position of the plane in world space.
+        /// </summary>
+        private Vector3 planePosition;
+
+        /// <summary>
+        /// Current distance of the plane from the user's head. Only used when not using the target override
+        /// or the GazeManager to set the plane's position. 
+        /// </summary>
+        private float currentPlaneDistance = 4.0f;
+
+        /// <summary>
+        /// Tracks the previous position of the target override object. Used if velocity is being tracked.
+        /// </summary>
+        private Vector3 targetOverridePreviousPosition;
+
+        /// <summary>
+        /// Used for representing latest plane drawn as gizmo
+        /// </summary>
+        private StabilizationPlaneOverride debugPlane;
+
+        /// <summary>
+        /// Used for representing the debug mesh
+        /// </summary>
+        private GameObject debugMesh;
+
+        private void Awake()
+        {
+#if UNITY_EDITOR
+            debugMesh = GameObject.CreatePrimitive(PrimitiveType.Quad);
+            debugMesh.gameObject.SetActive(false);
+#endif
+        }
+
+        /// <summary>
+        /// Updates the focus point for every frame after all objects have finished moving.
+        /// </summary>
+        private void LateUpdate()
+        {
+            float deltaTime = UseUnscaledTime
+                ? Time.unscaledDeltaTime
+                : Time.deltaTime;
+
+            switch (Mode)
+            {
+                case StabilizationPlaneMode.Fixed:
+                    ConfigureFixedDistancePlane(deltaTime);
+                    break;
+                case StabilizationPlaneMode.GazeHit:
+                    ConfigureGazeManagerPlane(deltaTime);
+                    break;
+                case StabilizationPlaneMode.PlaneOverride:
+                    ConfigureOverridePlane(deltaTime);
+                    break;
+                case StabilizationPlaneMode.TargetOverride:
+                    ConfigureTransformOverridePlane(deltaTime);
+                    break;
+                case StabilizationPlaneMode.Off:
+                default:
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Called by Unity when this script is loaded or a value is changed in the inspector.
+        /// Only called in editor, ensures that the property values always match the corresponding member variables.
+        /// </summary>
+        private void OnValidate()
```

Comment:
remove OnValidate, for any initialization/validation of values for a Monobehavior should be done in the Awake, OnEnable, or Start functions.

Reply:
We have had so many problems with onvalidate in past